pymeasure/instruments/oxfordinstruments/itc503.py

```diff
@@ -54,220 +75,417 @@ class ITC503(Instrument):
         print(itc.temperature_1)        # Print the temperature at sensor 1
 
     """
-    _T_RANGE = [0, 301]
+
+    def __init__(self,
+                 adapter,
+                 name="Oxford ITC503",
+                 clear_buffer=True,
+                 min_temperature=0,
+                 max_temperature=1677.7,
+                 **kwargs):
+
+        if isinstance(adapter, (int, str)):
+            kwargs.setdefault('read_termination', '\r')
+            kwargs.setdefault('send_end', True)
+            adapter = OxfordInstrumentsAdapter(
+                adapter,
+                asrl={
+                    'baud_rate': 9600,
+                    'data_bits': 8,
+                    'parity': 0,
+                    'stop_bits': 20,
+                },
+                preprocess_reply=lambda v: v[1:],
+                **kwargs,
+            )
+
```

Comment:
In my opinion, this approach will prevent users with a Prologix GPIB adapter to control the instrument, since part of the instrument behaviour is embedded in the `OxfordInstrumentsAdapter` which requires `VISAAdapter`.
Perhaps this is intended, but I just wanted to double check.
I think it is important to clarify the role of `Adapter` and `Instrument` classes (see also #567).

Reply:
I think we should at least try and see if we can get Prologix into VISA, see https://github.com/pymeasure/pymeasure/issues/567#issuecomment-1024996200, that would resolve that issue most neatly.

Reply:
If we succeed in having Prologix inside VISA (which is good), the entire `Adapter` classes are no longer needed, since all interfaces are handled by PyVISA. But, if we want to keep the `Adapter` classes, we need to define the scope and purpose of it. Of course, clarifying the role of `Adapter` is important anyway, regardless that Prologix is integrated or not in PyVISA.

Reply:
It was certainly not intended to restrict controlling this instrument to the `VISAAdapter` - I have to admit, I just started with it since I use pyvisa, and did not see this problem. I included my thoughts on this in [#567 (comment)](https://github.com/pymeasure/pymeasure/issues/567#issuecomment-1038857075).

Reply:
That's actually quite a good point. I'm not sure what is the best approach: should we try to fix this now, or hold this off until #567 is sorted out (in which case, should we wait with merging until it is resolved or not)?

Reply:
> I'd be inclined to leave it as it is now, merge it and modifying whatever's needed when https://github.com/pymeasure/pymeasure/issues/567 is sorted out.

Agreed. Other instruments also have the problem of not being able to use PrologixAdapter, which we will resolve with #567. Having this merged has the added benefit that a PR solving #567 has this instrument as a ready example in the codebase, to guide the implementation (as you immediately see how it works/"feels" in practice)

Reply:
I guess we could merge this now, since it conforms with the current way of doing things (which is sub-optimal as discussed in #567), and I think moving the custom adapter to a mixin class either on the `Adapter` or `Instrument` level will not be a lot of work, so we can then transition this one as an example, and add a prominent documentation of how to implement quirks of instruments/manufacturers.

Reply:
I agree with the proposal to merge, a bit of pragmatism over high level design discussions sometime is needed.
If the outcome of #567 will requires some further adaption, it could be done at that time, but for the time being people could enjoy the enhancements.

Reply:
Let's do it, then!

Reply:
Perfect, thanks!